File: c/reverse-list.c
/**
 * Tanggal: 5 Februari 2026
 * Link: https://www.codewars.com/kata/53da6d8d112bd1a0dc00008b/train/c
 */

#include <stddef.h>
#include <stdio.h>
#include <stdlib.h>

int *reverse_list(const int *numbers, size_t length) {
    int *result = (int *) malloc(sizeof(int) * length);

    for (int i = 0; i < length; i++) {
        result[length - 1 - i] = numbers[i];
    }

    return result;
}

int main() {
    const int numbers[] = {1, 2, 3, 4, 5};
    size_t length = 5;

    int *result = reverse_list(numbers, length);

    for (int i = 0; i < length; i++) {
        printf("%d\n", result[i]);
    }

    free(result);

    return 0;
}

File: php/reverse-list.php
<?php

# Tanggal: 13 Juli 2026
# Link: https://www.codewars.com/kata/53da6d8d112bd1a0dc00008b/train/php


function reverse_list(array $list) {
    $copy = $list;
    $half = ((int) (count($copy) / 2));

    for ($i = 0; $i < $half; $i++) {
        $target = count($copy) - $i - 1;
        $temp = $copy[$i];
        $copy[$i] = $copy[$target];
        $copy[$target] = $temp;
    }

    return $copy;
}

$result = reverse_list([1, 2, 3, 4, 5]);

foreach ($result as $item) {
    echo $item . PHP_EOL;
}
